client/src/marker.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Marker as LMarker, Popup } from 'react-leaflet';
+import { icon } from '../mapconfig';
+
+const Marker = (props) => {
+  const { detail, position, displayModal } = props;
```

Comment:
You can destructure props in function params. `({ detail, position, displayModal }) => ( rest of func)`